core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -211,6 +216,37 @@ private void processScanResult(Snapshot.Key key, Scan scan, TransactionResult re
     snapshot.putIntoReadSet(key, Optional.of(result));
   }
 
+  public TransactionCrudOperable.Scanner getScanner(Scan originalScan) throws CrudException {
+    List<String> originalProjections = new ArrayList<>(originalScan.getProjections());
+    Scan scan = (Scan) prepareStorageSelection(originalScan);
+
+    ConsensusCommitScanner scanner;
+
+    Optional<LinkedHashMap<Snapshot.Key, TransactionResult>> resultsInSnapshot =
+        snapshot.getResults(scan);
+    if (resultsInSnapshot.isPresent()) {
+      scanner =
+          new ConsensusCommitSnapshotScanner(scan, originalProjections, resultsInSnapshot.get());
+    } else {
+      scanner = new ConsensusCommitStorageScanner(scan, originalProjections);
+    }
```

Comment:
If the scan result of the same scan is in the scanSet in the snapshot, we use it. If not, we scan the storage directly.